src/v/kafka/server/replicated_partition.cc

```diff
@@ -486,7 +486,7 @@ ss::future<error_code> replicated_partition::prefix_truncate(
   model::offset kafka_truncation_offset,
   ss::lowres_clock::time_point deadline) {
     if (
-      kafka_truncation_offset <= start_offset()
+      kafka_truncation_offset < start_offset()
```

Comment:
I'm thinking we should short-circuit and return success directly if `kafka_truncation_offset == start_offset()`. Why would we replicate a no-op command?

Reply:
Added an early return for `kafka_truncation_offset == start_offset()`.